src/components/m-dropdown/index.js

```diff
@@ -1,60 +1,172 @@
 import PropTypes from '../../js/prop-types'; // eslint-disable-next-line import/first
-import classnames from 'classnames';
-
 import styles from './index.scss';
 import template from './_template';
 import BaseComponentGlobal from '../../js/abstract/base-component-global';
 import defineOnce from '../../js/define-once';
 import urlPropType from '../../js/prop-types/url-prop-type';
-import valuePropType from '../../js/prop-types/value-prop-type';
-import DropDown from './js/drop-down';
+import on from '../../js/on';
+import fire from '../../js/fire';
+import { EVENTS, AXA_EVENTS } from '../../js/ui-events';
 
+const DEFAULTS = {
+  selectClass: 'js-dropdown__content',
+  containerClass: '.js-dropdown',
+  toggleClass: 'js-dropdown__toggle',
+  nativeSelectClass: 'js-dropdown__native-select',
+  isOpenClass: 'is-dropdown-open',
+  isAnimatingClass: 'is-dropdown-animating',
+};
 
 class AXADropdown extends BaseComponentGlobal {
-  static tagName = 'axa-dropdown'
+  static tagName = 'axa-dropdown';
   static propTypes = {
-    inFlow: PropTypes.bool,
+    classes: PropTypes.string,
     items: PropTypes.arrayOf(PropTypes.shape({
       name: PropTypes.string,
       url: urlPropType,
-      value: valuePropType,
+      value: PropTypes.string,
+      isSelected: PropTypes.bool,
     })),
     native: PropTypes.bool,
     size: PropTypes.oneOf(['sm']),
     title: PropTypes.string,
-    value: valuePropType,
+    value: PropTypes.string,
+  }
+
+  static get observedAttributes() {
+    return ['items', 'title', 'native', 'value'];
   }
 
   init() {
-    super.init({ styles, template });
+    super.init({ styles, template }); // eslint-disable-next-line prefer-destructuring
+    this.selectedItem = this.items.filter(item => item.isSelected)[0];
   }
 
-  willRenderCallback() {
-    const { props: { inFlow, size, gpu } } = this;
+  connectedCallback() {
+    super.connectedCallback();
+    this.className = `${this.initialClassName} m-dropdown`;
+    this.isOpen = false; // use props.isOpen?
 
-    this.className = classnames(this.initialClassName, 'm-dropdown js-dropdown', {
-      'm-dropdown--in-flow': inFlow,
-      'm-dropdown--gpu': gpu,
-      [`m-dropdown--${size}`]: size,
-    });
+    this.onDropdownClick =
+      on(this, EVENTS.CLICK, DEFAULTS.toggleClass, this.handleDropdownClick, { capture: true, passive: false });
+
+    // the enhanced dropdown is a list with links -> clicks event. the native is a select change event
+    this.onDropdownValueClick =
+      on(this, EVENTS.CLICK, DEFAULTS.selectClass, e => this.handleDropdownValueClick(e), { capture: true, passive: false });
```

Comment:
please use class fields here too

Reply:
Oh. thanks. true, this fixes the IE 11 issues, too. Would be nice if you could explain this "magic"  behind there in an upcoming axa code review.

Reply:
done.

Reply:
Indeed, because it's rather obscure, I have even written it down:
https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTING.md#constructor